scripts/azureml-assets/azureml/assets/validate_assets.py

```diff
@@ -574,6 +595,188 @@ def validate_model_assets(latest_asset_config: assets.AssetConfig, validated_ass
         return 1
 
 
+def validate_model_spec(asset_config: assets.AssetConfig, validated_model_map: dict):
+    """Validate model spec."""
+    model = load_model(asset_config.spec_with_path)
+    if asset_config.type != assets.config.ModelType.MLFLOW:
+        logger.print(
+            f"Bypass validation for {asset_config.name} as model type is: {asset_config.type.value}"
+        )
+        return 0
+
+    # confirm must have tags
+    if not model.tags.get(MLFlowModelTags.TASK):
+        _log_error(asset_config.file_name_with_path, f"{MLFlowModelTags.TASK} missing")
+        return 1
+
+    if not model.tags.get(MLFlowModelTags.LICENSE):
+        _log_error(asset_config.file_name_with_path, f"{MLFlowModelTags.LICENSE} missing")
+        return 1
+
+    if not model.tags.get(MLFlowModelTags.INFERENCE_COMPUTE_ALLOWLIST):
+        _log_error(
+            asset_config.file_name_with_path,
+            (
+                f"{MLFlowModelTags.INFERENCE_COMPUTE_ALLOWLIST} missing. "
+                "Inference is the min requirement for model in system registry."
+            )
+        )
+        return 1
+
+    supports_eval = False
+    supports_ft = False
+
+    if not model.tags.get(MLFlowModelTags.EVALUATION_COMPUTE_ALLOWLIST):
+        supports_eval = True
+
+    if not model.tags.get(MLFlowModelTags.FINETUNE_COMPUTE_ALLOWLIST):
+        supports_ft = True
+
+    # check properties

```

Comment:
Looks like you have several areas like properties, valid computes, etc. which you could possibly separate into different functions

Reply:
thanks for suggestion @lykelly19. I've moved common compute validations to a new method.